src/DiagnosticMessages.ts

```diff
@@ -738,85 +891,99 @@ export let DiagnosticMessages = {
         data: {
             name: name,
             fullName: fullName ?? name,
             typeName: typeName ? typeName : undefined
         },
-        severity: DiagnosticSeverity.Error
+        severity: DiagnosticSeverity.Error,
+        code: 'cannot-find-function'
     }),
     argumentTypeMismatch: (actualTypeString: string, expectedTypeString: string, data?: TypeCompatibilityData) => ({
-        message: `Argument of type '${actualTypeString}' is not compatible with parameter of type '${expectedTypeString}'${typeCompatibilityMessage(actualTypeString, expectedTypeString, data)}`,
+        message: `Argument of type '${actualTypeString}' is not compatible with parameter of type '${expectedTypeString}'${typeCompatibilityMessage(actualTypeString, expectedTypeString, data)} `,
         data: data,
-        code: 1141,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1141,
+        severity: DiagnosticSeverity.Error,
+        code: 'argument-type-mismatch'
     }),
     returnTypeMismatch: (actualTypeString: string, expectedTypeString: string, data?: TypeCompatibilityData) => ({
-        message: `Type '${actualTypeString}' is not compatible with declared return type '${expectedTypeString}'${typeCompatibilityMessage(actualTypeString, expectedTypeString, data)}'`,
+        message: `Type '${actualTypeString}' is not compatible with declared return type '${expectedTypeString}'${typeCompatibilityMessage(actualTypeString, expectedTypeString, data)} '`,
         data: data,
-        code: 1142,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1142,
+        severity: DiagnosticSeverity.Error,
+        code: 'return-type-mismatch'
     }),
     assignmentTypeMismatch: (actualTypeString: string, expectedTypeString: string, data?: TypeCompatibilityData) => ({
         message: `Type '${actualTypeString}' is not compatible with type '${expectedTypeString}'${typeCompatibilityMessage(actualTypeString, expectedTypeString, data)}`,
         data: data,
-        code: 1143,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1143,
+        severity: DiagnosticSeverity.Error,
+        code: 'assignment-type-mismatch'
     }),
     operatorTypeMismatch: (operatorString: string, firstType: string, secondType = '') => ({
         message: `Operator '${operatorString}' cannot be applied to type${secondType ? 's' : ''} '${firstType}'${secondType ? ` and '${secondType}'` : ''}`,
-        code: 1144,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1144,
+        severity: DiagnosticSeverity.Error,
+        code: 'operator-type-mismatch'
     }),
     incompatibleSymbolDefinition: (symbol: string, scopeName: string) => ({
         message: `'${symbol}' is incompatible across these scopes: ${scopeName}`,
-        code: 1145,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1145,
+        severity: DiagnosticSeverity.Error,
+        code: 'incompatible-definition'
```

Comment:
```suggestion
        code: 'incompatible-type-definitions'
```

Reply:
After thinking about this more, I think I prefer the original message over the suggestion.